### PR TITLE
Make OxQL query response type into an object

### LINKED ISSUE #6371: Type returned from OxQL queries should be an object
This originally came up on this [docs page](https://github.com/oxidecomputer/docs/issues/431) issue, where the return type from an OxQL query isn't actually rendered. The docs page doesn't handle cases when the API response itself is an array type. We _have_ many arrays, but they're usually embedded in things like a `dropshot::ResultsPage` in the `items` field. The top-level return type is an object. We can do that from OxQL too, both for the docs page but also so we can add additional fields more easily if we need. I think returning something like the `QuerySummary`. We only use that internally today for understanding the resource consumption of the query, but some subset of it, say the latency, might be useful to expose in general. In any case, let's just wrap the existing array of `Table` into an object for now.

### PR DESCRIPTION
**Pull request by bnaecker**
- Fixes #6371

### CODE REVIEW

**Review by david-crespo — APPROVED** [review]
Looks good! I think with https://github.com/oxidecomputer/omicron/pull/6373, `OxqlQueryResult` will have to live in `nexus/types/src/external_api/views.rs`, so you might as well move it there already.

**Comment by bnaecker** [comment]
Makes sense, thanks. @sunshowers should I wait until the Nexus external API sequence lands to merge this? It's not a big conflict, but I don't want to hold the train.

File: nexus/src/external_api/http_entrypoints.rs
Comment (sunshowers) [inline nexus/src/external_api/http_entrypoints.rs:6392]: Could you use `views::OxqlQueryResult` here?

File: nexus/src/external_api/http_entrypoints.rs
Comment (sunshowers) [inline nexus/src/external_api/http_entrypoints.rs:6401]: And here.

**Review by sunshowers — COMMENTED** [review]
How about landing this after https://github.com/oxidecomputer/omicron/pull/6364? (That will cause a semantic merge conflict with this PR sadly.) The rest are pretty easy to fix up, thanks!

File: nexus/src/external_api/http_entrypoints.rs
Comment (bnaecker) [reply nexus/src/external_api/http_entrypoints.rs]: Yep, done in 42aa9cad6

File: nexus/src/external_api/http_entrypoints.rs
Comment (bnaecker) [reply nexus/src/external_api/http_entrypoints.rs]: Both done in 42aa9cad6

**Comment by bnaecker** [comment]
I've updated this on Rain's #6364, and set this to automerge. Thanks for the reviews y'all!.

**Pull request merged**
